### ISSUE TITLE
[CHE 6] Can't login to multiuser Che server deployed on OCP or minishift with default admin credentials

### ISSUE THREAD

**Issue by sleshchenko** [eclipse-che/che#7131]
### Description
Can't login to multiuser Che server deployed on `OCP` or `minishift` with default admin credentials.

### Reproduction Steps
1. Deploy Che server on `minishift` or `OCP`.
2. Try to login with default admin credentials. I suppose it is `admin` value for login and password.
Expected: User is logged in.
Actuals: Failed to log in.
![screenshot_20171101_162451](https://user-images.githubusercontent.com/5887312/32279361-37a16d22-bf21-11e7-9ef4-720bb379fba7.png)

Note that is it possible to login to `Keycloak Admin` console with `admin:admin` credentials and Che Realms doesn't have any users.
As far as I know, it works fine for Che server started by `CLI`.
![screenshot_20171101_162417](https://user-images.githubusercontent.com/5887312/32279337-250f0c96-bf21-11e7-8c0c-341e4c393c54.png)

**Comment by riuvshin** [comment]
we decided to try "replace existing" users policy because it looks like db is already provisioned somehow.

**Comment by riuvshin** [comment]
after some investigation we've figured that KC deployed to ocp / minishift starts with some stragene messages in logs:
```
10:10:43,503 INFO  [org.keycloak.exportimport.dir.DirImportProvider] (ServerService Thread Pool -- 59) Importing from directory /opt/jboss/keycloak/realms
--
  | 10:10:43,714 INFO  [org.keycloak.services] (ServerService Thread Pool -- 59) KC-SERVICES0030: Full model import requested. Strategy: IGNORE_EXISTING
  | 10:10:43,847 INFO  [org.keycloak.exportimport.util.ImportUtils] (ServerService Thread Pool -- 59) Realm 'master' already exists. Import skipped
  | 10:10:43,939 INFO  [org.keycloak.exportimport.util.ImportUtils] (ServerService Thread Pool -- 59) Realm 'che' already exists. Import skipped
  | 10:10:43,952 INFO  [org.keycloak.services] (ServerService Thread Pool -- 59) KC-SERVICES0032: Import finished successfully
```

That happen even after minishift delete which destroy whole VM and there is no chance to keep some db files. Which probably means that issue somewhere in how we deploy KC / Postgress to minishift or ocp


That issue does not happen with che multiuser on docker.


@davidfestal any idea why this happen?

**Comment by riuvshin** [comment]
I've found errors in very first KC server deployment
```
12:59:20,182 ERROR [org.jboss.msc.service.fail] (ServerService Thread Pool -- 60) MSC000001: Failed to start service jboss.undertow.deployment.default-server.default-host./auth: org.jboss.msc.service.StartException in service jboss.undertow.deployment.default-server.default-host./auth: java.lang.RuntimeException: RESTEASY003325: Failed to construct public org.keycloak.services.resources.KeycloakApplication(javax.servlet.ServletContext,org.jboss.resteasy.core.Dispatcher)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:84)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at org.jboss.threads.JBossThread.run(JBossThread.java:320)
Caused by: java.lang.RuntimeException: RESTEASY003325: Failed to construct public org.keycloak.services.resources.KeycloakApplication(javax.servlet.ServletContext,org.jboss.resteasy.core.Dispatcher)
	at org.jboss.resteasy.core.ConstructorInjectorImpl.construct(ConstructorInjectorImpl.java:162)
	at org.jboss.resteasy.spi.ResteasyProviderFactory.createProviderInstance(ResteasyProviderFactory.java:2298)
	at org.jboss.resteasy.spi.ResteasyDeployment.createApplication(ResteasyDeployment.java:340)
	at org.jboss.resteasy.spi.ResteasyDeployment.start(ResteasyDeployment.java:253)
	at org.jboss.resteasy.plugins.server.servlet.ServletContainerDispatcher.init(ServletContainerDispatcher.java:120)
	at org.jboss.resteasy.plugins.server.servlet.HttpServletDispatcher.init(HttpServletDispatcher.java:36)
	at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:117)
	at org.wildfly.extension.undertow.security.RunAsLifecycleInterceptor.init(RunAsLifecycleInterceptor.java:78)
	at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:103)
	at io.undertow.servlet.core.ManagedServlet$DefaultInstanceStrategy.start(ManagedServlet.java:250)
	at io.undertow.servlet.core.ManagedServlet.createServlet(ManagedServlet.java:133)
	at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:565)
	at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:536)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1508)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1508)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1508)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1508)
	at io.undertow.servlet.core.DeploymentManagerImpl.start(DeploymentManagerImpl.java:578)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:100)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:81)
	... 6 more
Caused by: java.lang.RuntimeException: Error during export/import: Unexpected character ('<' (code 60)): was expecting double-quote to start field name
 at [Source: java.io.FileInputStream@e108c85; line: 26, column: 6]
```

So the reason is why it happen is absolute misusing of puppet templates here: https://github.com/eclipse/che/blob/che6/dockerfiles/init/s2i/keycloak/assemble#L42 these files supposed to be templates that generate actual config files with given set of params. 
@l0rd I have no idea yet how to handle that properly... I will try to do same dirty hack with `sed` to replace another injections here: 
https://github.com/eclipse/che/blob/che6/dockerfiles/init/modules/keycloak/templates/che-users-0.json.erb#L26
But this is not something I really want to do... 
I'm not that much familiar with s2i but this approach does not work for us. @l0rd @benoitf if you huys have any idea how to properly handle that please share :)

**Comment by davidfestal** [comment]
@riuvshin as I talked to you on BJ when we looked into this issue, initially the realm `.erb` files didn't have anything to replace. This is why I used them directly in the s2I build. And when I had to introduce a variable to manage the che-server url, it was much simpler to use a `sed` to replace this value in the s2i build.

I agree that, if more complex puppet syntax is used (such as your last change), we should probably change the way it is done. However, it seems to me that this could be part of the work on a CLI command that would allow deploying to Openshift.

However if that can help, I can take a look, to find a better way to manage this.
